docs/rules/component-tags-order.md

```diff
@@ -0,0 +1,142 @@
+---
+pageClass: "rule-details"
+sidebarDepth: 0
+title: "svelte/component-tags-order"
+description: "Enforce order of component top-level elements"
+---
+
+# svelte/component-tags-order
+
+> Enforce order of component top-level elements
+
+- :exclamation: <badge text="This rule has not been released yet." vertical="middle" type="error"> **_This rule has not been released yet._** </badge>
+- :wrench: The `--fix` option on the [command line](https://eslint.org/docs/user-guide/command-line-interface#fixing-problems) can automatically fix some of the problems reported by this rule.
+
+## :book: Rule Details
+
+This rule warns about the order of the top-level tags, such as `<script>`, template and `<style>`.
+
+<ESLintCodeBlock fix>
+
+<!--eslint-skip-->
+
+```svelte
+<svelte:options />
+
+<!-- ✓ GOOD -->
+
+<script context="module">
+  /* eslint svelte/component-tags-order: "error" */
+</script>
+
+<script>
+  export let foo
+</script>
+
+<svelte:window />
+<svelte:document />
+<svelte:head />
+<svelte:body />
+<div />
+
+<style>
+  p {
+    color: blue;
+  }
+</style>
+```
+
+</ESLintCodeBlock>
+
+---
+
+<ESLintCodeBlock fix>
+
+<!-- prettier-ignore-start -->
+<!--eslint-skip-->
+
+```svelte
+<!-- ✗ BAD -->
+
+<script>
+  /* eslint svelte/component-tags-order: "error" */
+  export let foo
+</script>
+
+<script context="module">
+  const bar = 42
+</script>
+```
+
+<!-- prettier-ignore-end -->
+
+</ESLintCodeBlock>
+
+---
+
+<ESLintCodeBlock fix>
+
+<!-- prettier-ignore-start -->
+<!--eslint-skip-->
+
+```svelte
+<!-- ✗ BAD -->
+<div>{foo}</div>
+
+<style>
+  div {
+    color: blue;
+  }
+</style>
+
+<script>
+  /* eslint svelte/component-tags-order: "error" */
+  export let foo
+</script>
+
+```
+
+<!-- prettier-ignore-end -->
+
+</ESLintCodeBlock>
+
+## :wrench: Options
+
+```json
+{
+  "svelte/component-tags-order": [
+    "error",
+    {
+      "order": [
+        "SvelteScriptElement([context=module])",
```

Comment:
Hmm. I don't think it's a good idea to use a format specifically for this rule.
For example, using the CSS selector format most users already know the format. (However, users may be a little surprised because the colon must be escaped. e.g. `svelte\:options`)
Is the format currently used by the rule inspired by something?

Reply:
I copied most parts from https://eslint.vuejs.org/rules/component-tags-order.html
And while implementing the rule, I felt that we need to support `and` condition, but I couldn't find the way to use `and` condition in the ESLint selector.
https://eslint.org/docs/latest/extend/selectors

And I think that the ESLint selector is complicated for most users because they need to understand AST.

That's why I go to use a special format.
(But yeah, I didn't support escaping things...)

Reply:
> And while implementing the rule, I felt that we need to support `and` condition, but I couldn't find the way to use `and` condition in the ESLint selector.

Are you saying we can't use `and` in CSS selectors?
I think that the and condition in the CSS selector just connects without using spaces. e.g. `script[context=module][lang=ts]`
And in modern CSS selectors the `or` condition uses `:is()`. e.g. `:is(script, style)`


> https://eslint.org/docs/latest/extend/selectors

I agree that esquery is difficult for users to understand.

Reply:
By the way, eslint-plugin-vue implements a function that converts CSS selectors to matchers in order to use CSS selectors in defining [`vue/component-tags-order`](https://eslint.vuejs.org/rules/component-tags-order.html) rule option.



https://github.com/vuejs/eslint-plugin-vue/blob/81ce0cecb87eae60830b9b973790d4f2386062c1/lib/utils/selector.js#L22